app/src/main/java/com/perflyst/twire/activities/main/MyStreamsActivity.java

```diff
@@ -51,18 +56,41 @@ public void addToAdapter(List<StreamInfo> aObjectList) {
     }
 
     @Override
-    public List<StreamInfo> getVisualElements() throws JSONException, MalformedURLException {
-        final String URL = "https://api.twitch.tv/kraken/streams/followed?oauth_token=" + new Settings(getBaseContext()).getGeneralTwitchAccessToken() + "&limit=" + getLimit() + "&offset=" + getCurrentOffset() + "&stream_type=live";
-        final String ARRAY_KEY = "streams";
+    public List<StreamInfo> getVisualElements() throws JSONException, ExecutionException, InterruptedException, MalformedURLException {
+        // build the api link
+        String helix_url = "https://api.twitch.tv/helix/streams?first=100";
+
+        ArrayList<ChannelInfo> channels;
+        if (TempStorage.hasLoadedStreamers()) {
+            channels = new ArrayList<>(TempStorage.getLoadedStreamers());
+        } else {
+            GetFollowsFromDB subscriptionsTask = new GetFollowsFromDB();
+            subscriptionsTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR, getBaseContext());
+            channels = new ArrayList<>(subscriptionsTask.get().values());
+        }
 
+        // loop over all the channels in the DB in chunks of 100
         List<StreamInfo> mResultList = new ArrayList<>();
-        String jsonString = Service.urlToJSONString(URL);
-        JSONObject fullDataObject = new JSONObject(jsonString);
-        JSONArray topStreamsArray = fullDataObject.getJSONArray(ARRAY_KEY);
-
-        for (int i = 0; i < topStreamsArray.length(); i++) {
-            JSONObject streamObject = topStreamsArray.getJSONObject(i);
-            mResultList.add(JSONService.getStreamInfo(getBaseContext(), streamObject, null, false));
+        final String ARRAY_KEY = "data";
+        for (List<ChannelInfo> chunk : Lists.partition(channels, 100)) {
+            // this returns a list, I found no easy way to convert it to a string
+            List ids = Lists.transform(chunk, channelInfo -> "&user_id=" + channelInfo.getUserId());
+            // convert the new list to a string
+            String id_string = "";
+            for (int i=0; i<ids.size(); i++) {
+                id_string = id_string + ids.get(i);
+            }
+            String url = helix_url + id_string;
```

Comment:
Oh sorry about that, there is a easy way to make the list into a string. I forgot to use that in my suggestion.
```suggestion
            String url = helix_url + Joiner.on("").join(Lists.transform(chunk, channelInfo -> "&user_id=" + channelInfo.getUserId()));
```

Reply:
Thanks! Changed in c500ea5546059102be290b92bcced70a51ca3a90